File: main/src/main/java/lavaplayer/source/youtube/YoutubeSearchResultLoader.java
package lavaplayer.source.youtube;

import lavaplayer.tools.http.ExtendedHttpConfigurable;
import lavaplayer.track.AudioItem;
import lavaplayer.track.AudioTrack;
import lavaplayer.track.AudioTrackInfo;

import java.util.function.Function;

public interface YoutubeSearchResultLoader {
    AudioItem loadSearchResult(String query, Function<AudioTrackInfo, AudioTrack> trackFactory);

    ExtendedHttpConfigurable getHttpConfiguration();
}

File: main/src/main/java/lavaplayer/source/youtube/YoutubeSearchResultLoader.kt
package lavaplayer.source.youtube

import lavaplayer.tools.http.ExtendedHttpConfigurable
import lavaplayer.track.AudioItem
import lavaplayer.track.AudioTrackFactory

interface YoutubeSearchResultLoader {
    val httpConfiguration: ExtendedHttpConfigurable

    fun loadSearchResult(query: String, trackFactory: AudioTrackFactory): AudioItem
}
